dev/src/index.ts

```diff
@@ -987,12 +994,39 @@ export class Firestore {
     );
   }
 
+  /**
+   * Registers the requestTag associated with the listener. This is used to
+   * ensure that all listeners are detached when terminate() is called.
+   *
+   * @private
+   * @param requestTag A unique client-assigned identifier
+   */
+  registerListener(requestTag: string): void {
```

Comment:
It would be nice if this was somehow scoped to the class itself, but I can't find a good way to do this. `readStream` would be a natural place, but it would allow stream retries after the client is terminated.

Reply:
I spent a chunk of time thinking about this last week, and I couldn't think of a way to pipe it through another way. 
 - `ClientPool` is charge of terminating the client.
 - It seems like the only place to accurately track listeners w/o having to deal with extra node stream logic is to monitor each listener registration in watch.
- `Firestore` is the only place to store the Watch information.